File: repos_estudiantes_ALVAREZ_primera_entrega/CARDON_PAZ_LUIS_ABDEL/POLIMORFISMO/EJERCICIO_5/Celular.java
package POLIMORFISMO.EJERCICIO_5;

public class Celular {
    private String nroTel;
    private String dueño;
    private double espacio;
    private int ram;
    private int nroApp;
    
    public Celular(String nroTel, String dueño, double espacio, int ram, int nroApp) {
        this.nroTel = nroTel;
        this.dueño = dueño;
        this.espacio = espacio;
        this.ram = ram;
        this.nroApp = nroApp;
    }
    
    public Celular operatorPlusPlus() {
        this.nroApp += 10;
        return this;
    }
    
    public Celular operatorMinusMinus() {
        this.espacio -= 5;
        return this;
    }
    
    public void mostrarDatos() {
        System.out.println("Numero: " + nroTel);
        System.out.println("Dueño: " + dueño);
        System.out.println("Espacio: " + espacio + " GB");
        System.out.println("RAM: " + ram + " GB");
        System.out.println("Numero de Apps: " + nroApp);
        System.out.println("------------------------");
    }
}

File: repos_estudiantes_ALVAREZ_primera_entrega/APAZA_PACHECO_BRYAN_ORLANDO/Polimorfismo/Ejer5/Celular.py
# a) Class
class Celular:
    # SE DEBE SOBRECARGAR EL OPERADOR
    # a) 
    def __init__(self,nroTel,dueño,espacio,ram,nroApp):
        self.nroTel=nroTel
        self.dueño=dueño
        self.espacio=espacio
        self.ram=ram
        self.nroApp=nroApp

    # b)
    # SE DEBE SOBRECARGAR LOS OPERADORES ++ - + 
    def aumentarApps(self):
        self.nroApp+=10

    # c) 
    def disminuirEspacio(self):
        self.espacio-=5

    # d) 
    def mostrarDatos(self):
        print("Tel:",self.nroTel,"Dueño:",self.dueño,"Espacio:",self.espacio,"GB","RAM:",self.ram,"GB","Apps:",self.nroApp)

# d) 
print("///////////////")
cel=Celular("78945612","Bryan",64,8,15)
print("Datos antes de operadores:")
cel.mostrarDatos()
cel.aumentarApps()
cel.disminuirEspacio() 
print("//////////////")
print("Datos después de operadores:")
cel.mostrarDatos()
